electronics_lib/PowerConditioning.py

```diff
@@ -263,3 +263,107 @@ def connected_from(self, gnd: Optional[Port[VoltageLink]] = None, pwr_hi: Option
     if pwr_lo is not None:
       cast(Block, builder.get_enclosing_block()).connect(pwr_lo, self.pwr_lo)
     return self
+
+
+class SoftPowerGate(PowerSwitch, KiCadSchematicBlock, Block):  # migrate from the multimater
+  """A high-side PFET power gate that has a button to power on, can be latched
+  on by an external signal, and provides the button output as a signal.
+
+  Attributes:
+    pull_resistance: The resistance of the pull-up resistor.
+    amp_resistance: The resistance of the amplifier resistor.
+    diode_drop: The voltage drop across the diode.
+
+  Ports:
+    pwr_in: A voltage sink port for the input power.
```

Comment:
Hm, stylistically I might omit the port type information (since it's redundant with the code). Or if you think the type information is important, maybe use the port type name directly, eg `VoltageSink`?

Reply:
I think btn_in/btn_out/control are better to have descriptions, and I have other ports for the sake of completeness. Then, update to use the type name directly like suggested?

Reply:
Also this is google styled docsting but if you have other preference i can change it